etc/nginx/nginx.conf

```diff
@@ -57,22 +58,38 @@ http {
             try_files $uri $uri/ /index.php?$args;
         }
 
-        location ~* \.(?:ico|css|js|gif|jpe?g|png|ttf|woff)$ {
+        location ~* \.(?:ico|gif|jpe?g|png|ttf|woff)$ {
             access_log off;
             expires 30d;
             add_header Pragma public;
             add_header Cache-Control "public, mustrevalidate, proxy-revalidate";
         }
-        
+
+        location ~ ^/(protected|framework|themes/\w+/views|\.|uploads/file) {
+            deny all;
+        }
+
+        location ~ ^/(assets|static|themes|uploads) {
+            expires 10d;
```

Comment:
as mentioned here we should also take args into account to avoid breaking on updates: https://docs.humhub.org/docs/admin/reverse-proxy#caching